hydra/_internal/hydra.py

```diff
@@ -90,14 +91,19 @@ def run(
             run_mode=RunMode.RUN,
         )
 
-        return run_job(
+        ret = run_job(
             config=cfg,
             task_function=task_function,
             job_dir_key="hydra.run.dir",
             job_subdir_key=None,
             configure_logging=with_log_configuration,
         )
 
+        if ret.job_result == JobResult.FAILED:
+            raise ret._return_value
```

Comment:
can you show an example exception printout for this case?

Reply:
Use `ret.return_value()`

Since it's a property, you can probably just replace this with 
```python
return ret.return_value
```